modules/system/assets/js/framework.js

```diff
@@ -479,6 +508,68 @@ if (window.jQuery.request !== undefined) {
         return this
     }
 
+    // $.fn.getParentFormData() PLUGIN DEFINITION
+    // ==============
+
+    $.fn.getParentFormData = function () {
+        // Identify and merge the data from parent form elements
+        //
+        // If the current $form element was spawned by an element that exists in a different form element
+        // (usually because the current form is in a popup), then it will have a data-request-parent
+        // attribute present that identifies the element that spawned it. We then need to merge the data
+        // from that element's containing form element and in turn check to see if it was spawned by another
+        // element (i.e. nested popups) until we have completed the chain down to the original form element.
+        // The closer a form element is to the current request $el, the higher priority its data will be in
+        // merge conflicts. This is required in order to ensure that all the data that was required to render
+        // a given widget is still present in any requests made to that widget so that it will be properly
+        // instantiated on the server side and ready to respond to our requests.
+        var $form = $(this).first(),
+            parentDataObjects = [serializeFormToObj($form)],
+            parentFormData = {};
+
+        var findParentForms = function ($form) {
+            // If the form element exists and has a parent element defined
+            if ($form.length && $form.data('request-parent')) {
+                // Identify the owning form of the parent element
+                var $parentEl = $($form.data('request-parent'));
+                if ($parentEl.length) {
+                    var parentEmbeddedData = {};
```

Comment:
@LukeTowers unless I'm mistaken, this seems to be a dangling reference. It's extended upon in the next couple of lines, but it's not returned or added to a parent variable?

Reply:
That would be correct, it needs to be added to the merge stack, will figure out what order it needs to be added in tomorrow.